sklearn/tree/tests/test_tree.py

```diff
@@ -1614,7 +1649,8 @@ def test_invalid_presort(cls):
     msg = ("'presort' should be in {}. "
            "Got {!r} instead.".format(allowed_presort, invalid_presort))
     est = cls(presort=invalid_presort)
-    assert_raise_message(ValueError, msg, est.fit, X, y)
+    with pytest.raises(ValueError, match=msg):
```

Comment:
this one fails because:

```
The regexp parameter of the match method is matched with the re.search function, so in the above example match='123' would have worked as well.
```
whereas before it was a simple string comparison, I think.

Reply:
@adrinjalali So can you tell me the correct way of writing this case using `pytest.raises`?

Reply:
In this case:

```python suggestion
    with pytest.raises(ValueError, match=msg.replace('(', '\(').replace(')', '\)')):
```

would work, but I'm not sure if it's the most elegant solution.

Reply:
Indeed a bit cumbersome, but nothing better comes immediatiely to mind.